backend/api/migrations/0056_alter_computetask_status.py

```diff
@@ -0,0 +1,51 @@
+# Generated by Django 4.2.9 on 2024-02-13 14:38
+
+from django.db import migrations
+from django.db import models
+
+
+def migrate_data(apps, schema_editor):
+    status_mapping = {
+        "STATUS_WAITING": "STATUS_WAITING_FOR_PARENT_TASKS",
+        "STATUS_TODO": "STATUS_WAITING_FOR_EXECUTOR_SLOT",
+    }
+    model_model = apps.get_model("api", "computetask")
+    for task_instance in model_model.objects.all():
+        new_status = status_mapping.get(task_instance.status_old, task_instance.status_old)
+        task_instance.category = new_status
+        task_instance.save()
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("api", "0055_function_image_address_function_image_checksum"),
+    ]
+
+    operations = [
+        migrations.RenameField(
+            model_name="computetask",
+            old_name="status",
+            new_name="status_old",
+        ),
+        migrations.AddField(
+            model_name="computetask",
+            name="status",
+            field=models.CharField(
+                choices=[
+                    ("STATUS_WAITING_FOR_BUILDER_SLOT", "Status Waiting For Builder Slot"),
+                    ("STATUS_BUILDING", "Status Building"),
+                    ("STATUS_WAITING_FOR_PARENT_TASKS", "Status Waiting For Parent Tasks"),
+                    ("STATUS_WAITING_FOR_EXECUTOR_SLOT", "Status Waiting For Executor Slot"),
+                    ("STATUS_DOING", "Status Doing"),
+                    ("STATUS_DONE", "Status Done"),
+                    ("STATUS_CANCELED", "Status Canceled"),
+                    ("STATUS_FAILED", "Status Failed"),
+                ],
+                max_length=64,
+                default="STATUS_DONE",
+            ),
+            preserve_default=False,
+        ),
+        migrations.RunPython(migrate_data),
```

Comment:
no reverse data migration?

Reply:
I didn't see any down migration, either for data or schema. Should we start adding it?

Reply:
IMO, it depends how much effort it is 🤷 
If it's more or less free, it's always nice to have, in case we need to revert on a production project. Else, if it's not the case for other migrations anyway, don't spend time on it.

Reply:
I don't even find documentation on reverse data_migration and migrations [12](https://github.com/Substra/substra-backend/blob/ef127d130c2ce3544131c8637ceea9c7df7ed040/backend/api/migrations/0012_alter_algo_category.py#L46) and [14](https://github.com/Substra/substra-backend/blob/ef127d130c2ce3544131c8637ceea9c7df7ed040/backend/api/migrations/0014_alter_model_category.py#L42) don't include it